### PR TITLE
semantic-release: init at 21.0.5

### PR DESCRIPTION
**Pull request by sestrella**
###### Description of changes

Introduces a new package called [semantic-release](https://github.com/semantic-release/semantic-release).

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by natsukium** [comment]
Would you please add it using `buildNpmPackage`?
We don't want to add new packages to `node-packages` as much as possible.

see, #229475

**Comment by sestrella** [comment]
@ehllie done! I used https://github.com/NixOS/nixpkgs/pull/234166 as a reference. Additionally, I wonder if it would make sense to open a follow-up PR updating the instructions in the JavaScript [guideline](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/javascript.section.md) under the "Adding and Updating Javascript packages in nixpkgs" section to be more specific, I followed the instructions here to submit the original changes. I'll be more than happy to submit a PR updating the docs

File: pkgs/development/node-packages/aliases.nix [hunk 1]
```diff
@@ -44,6 +44,7 @@ mapAliases {
   "@nestjs/cli" = pkgs.nest-cli; # Added 2023-05-06
   eslint_d = pkgs.eslint_d; # Added 2023-05-26
   manta = pkgs.node-manta; # Added 2023-05-06
+  semantic-release = pkgs.semantic-release; # Added 2022-06-09
```
Comment (ehllie) [inline pkgs/development/node-packages/aliases.nix:47]: I don't think it's necessary to include the package in the `aliases.nix` file, if it was never in the `nodePackages` attribute set in the first place. It should be there just for backwards compatibility and possibly deprecation at some point in the future.

**Review by ehllie — CHANGES_REQUESTED** [review]

File: pkgs/development/node-packages/aliases.nix (hunk 1 above)
Comment (sestrella) [reply pkgs/development/node-packages/aliases.nix]: @ehllie thank you for the explanation. I removed the package from `aliases.nix`

File: pkgs/development/tools/semantic-release/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{ cctools, buildNpmPackage, fetchFromGitHub, lib, python3, stdenv }:
+
+buildNpmPackage rec {
+  pname = "semantic-release";
+  version = "21.0.3";
+
+  src = fetchFromGitHub {
+    owner = "semantic-release";
+    repo = pname;
```
Comment (AndersonTorres) [inline pkgs/development/tools/semantic-release/default.nix:9]:
```suggestion
    repo = "semantic-release";
```

File: pkgs/development/tools/semantic-release/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      '"version": "0.0.0-development"' \
+      '"version": "${version}"'
+  '';
+
+  meta = with lib; {
+    description = "Fully automated version management and package publishing";
+    homepage = "https://semantic-release.gitbook.io/semantic-release/";
+    license = licenses.mit;
+    maintainers = [ maintainers.sestrella ];
+  };
```
Comment (AndersonTorres) [inline pkgs/development/tools/semantic-release/default.nix:34]:
```suggestion
  meta = {
    description = "Fully automated version management and package publishing";
    homepage = "https://semantic-release.gitbook.io/semantic-release/";
    license = lib.licenses.mit;
    maintainers = [ lib.maintainers.sestrella ];
  };
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Reorder the commits, _as explained in the_ [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) _you marked as fitting_.

**Comment by natsukium** [comment]
@sestrella san, thanks for the change.
> Additionally, I wonder if it would make sense to open a follow-up PR updating the instructions in the JavaScript [guideline](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/javascript.section.md) under the "Adding and Updating Javascript packages in nixpkgs" section to be more specific

Yes, it looks like this part needs to be changed.
I am unfamiliar with the documentation, but I think starting with an issue would be a good idea.
There needs to be a step to discuss the direction of the proposed improvements with the node team and other related folks.

File: pkgs/development/tools/semantic-release/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{ cctools, buildNpmPackage, fetchFromGitHub, lib, python3, stdenv }:
```
[the remark this answers is not in the document]
Comment (sestrella) [reply pkgs/development/tools/semantic-release/default.nix]: @natsukium thank you, I reformatted this file accordingly

File: pkgs/development/tools/semantic-release/default.nix [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+  npmDepsHash = "sha256-0GhDXGyVoMwz+vVFdplv7ObTiTRTvQDd9r9e6616iTI=";
+
+  dontNpmBuild = true;
+
+  nativeBuildInputs = [
+    python3
+  ] ++ lib.optional stdenv.isDarwin cctools;
+
+  # Fixes `semantic-release --version` output
```
Comment (AndersonTorres) [inline pkgs/development/tools/semantic-release/default.nix:28]: It is a bit funny that a package about semantic version needs a fixup on semantic version...

**Review by AndersonTorres — APPROVED** [review]
Ping me 2 days

**Review by natsukium — APPROVED** [review]
It works fine on `x86_64-linux` too.

**Comment by AndersonTorres** [comment]
I am waiting Darwin ARM.

File: pkgs/development/tools/semantic-release/default.nix (hunk 5 above)
Comment (sestrella) [reply pkgs/development/tools/semantic-release/default.nix:28]: Yeah, although, I think it is done this way so the tool itself takes care of updating the version during a release process

**Pull request merged**